SCons/Util.py

```diff
@@ -1420,32 +1449,24 @@ def f(self, x, y):
     if name is None:
         name = function.__name__
     else:
-        function = RenameFunction(function, name)
+        # "rename"
+        function = FunctionType(
+            function.__code__, function.__globals__, name, function.__defaults__
+        )
 
-    # Note the Python version checks - WLB
-    # Python 3.3 dropped the 3rd parameter from types.MethodType
     if hasattr(obj, '__class__') and obj.__class__ is not type:
-        # "obj" is an instance, so it gets a bound method.
-        if sys.version_info[:2] > (3, 2):
-            method = MethodType(function, obj)
+        # obj is an instance, so it gets a bound method.
+        if hasattr(obj, "added_methods"):
```

Comment:
So this is the meat of the change?

Reply:
yes.. if it "looks like" an env, handle a little differently